src/session/session-utils.c

```diff
@@ -70,23 +63,74 @@ read_authorize_response (const char *what)
        exit (0);
     }
 
+  /* make room for and add null terminator */
+  message = realloc (message, len + 1);
+  message[len] = '\0';
+  return (char *)message;
+}
+
+static const char *skip_quoted_value (const char *str)
+{
+  if (!*str || *str++ != '"')
+    errx (EX, "syntax error, expected quoted value");
+  while (*str && *str != '"')
+    str++;
+  if (*str != '"')
+    errx (EX, "syntax error, expected closing quote");
+  return str + 1;
+}
+
+char *get_authorize_key (const char *json, const char *key, bool required)
+{
   /*
-   * The authorize messages we receive always have an exact prefix and suffix:
+   * The authorize messages we receive always have this structure:
+   *
+   * \n{"command":"authorize","cookie":"NNN","response":"....","remote-peer":"..."}
```

Comment:
Hilariously trivial nag: at this point in the commit history, the `"remote-peer"` key doesn't exist yet :)

Reply:
Right! Fixed.